trin-core/src/portalnet/overlay_service.rs

```diff
@@ -119,6 +119,10 @@ pub enum OverlayRequestError {
     #[error("Invalid response")]
     InvalidResponse,
 
+    /// An invalid response was received.
+    #[error("Response failed validation")]
+    FailedValidation,
```

Comment:
IMO it will be nice if we include the actual content value in this error message. If validation fails, it will be useful to know the content value that we are receiving.